src/record/record.js

```diff
@@ -345,10 +324,13 @@ Record.prototype._recoverRecord = function( remoteVersion, remoteData, message )
  * @private
  * @returns {void}
  */
-Record.prototype._onRecordRecovered = function( remoteVersion, error, data ) {
+Record.prototype._onRecordRecovered = function( remoteVersion, remoteData, error, data ) {
 	if( !error ) {
 		this.version = remoteVersion;
-		this.set( data );
+		if ( data !== this._$data && !utils.deepEquals( data, this._$data ) ) {
```

Comment:
I'm confused. Code on master does not have a check..

https://github.com/deepstreamIO/deepstream.io-client-js/blob/master/src/record/record.js#L351